src/cmd/run.rs

```diff
@@ -104,11 +108,42 @@ pub async fn main<D: Database>(ctx: &AppContext, config: Config, db: D) -> Resul
         db,
     };
 
+    for integration in ctx.config.integrations().iter() {
+        if let Err(err) = validate_integration_configuration(&ctx.db, integration).await {
+            warn!("Unable to validate '{integration}': {err:#?}");
+        }
+    }
+
     let healthcheck_worker = healthcheck(&ctx.db);
     let integration_worker = integrations(&ctx);
     try_join!(healthcheck_worker, integration_worker).discard_ok()
 }
 
+#[tracing::instrument(skip_all)]
+async fn validate_integration_configuration<D: Database>(
+    db: &D,
+    integration: &Integration,
+) -> Result<(), Error> {
+    let repository = integration.remote().for_coordinate();
+    let import_branches = integration.import_branches();
+    let import_tags = integration.import_tags();
+
+    if !*import_branches {
+        let is_branch = true;
+        db.delete_states(&repository, is_branch)
+            .await
+            .change_context(Error::TaskDeleteState)?;
+    }
+    if !*import_tags {
+        let is_branch = false;
+        db.delete_states(&repository, is_branch)
+            .await
+            .change_context(Error::TaskDeleteState)?
+    }
```

Comment:
The difference here is the value of is_branch. Is_branch being set to true will remove all branch entries in the db. When is_branch is set to false in the function call, all tag entries in the db will be removed.

With that being said I can still combine the two if statements and have a ternary operator to deduce the value of is_branch. I just thought that separating it into two different if statements would make the intended action more clear and readable

Reply:
I can see what you're saying now and agree that they can't be combined without adding a special case for when both are `False`. 

I'd strongly recommend having some comments like [jess](https://github.com/fossas/broker/pull/134#discussion_r1336238985) asked for here too. Tbh I totally missed the fact that `is_branch` was a different value for each `if` statement when I first read this. I should have looked closer, but I think if I understood the intent of this function better I would have been able to maybe infer why you needed to handle it this way.

What would you think about changing some names around? To me, the name `validate_integration_configuration` implies checking rather than modifying data the way that this function does. I generally would expect a function that validates to return a value and not to do a side-effecting operation like deleting database rows. Would something like `configure_repository_scan_targets` fit as a name that describes what's really happening here?

Reply:
Totally valid. I will add some comments and  with the addition of changing the bools to an enum, I think it will make the intention a lot more clear. 

What do you think of these names for the function `clear_repository_scan_targets` or `remove_repository_scan_targets`?

Reply:
I think those are better than what I suggested since you're only deleting here.